### ISSUE TITLE
`redundant_sendable` should not apply to protocols

### ISSUE THREAD

**Issue by riley-williams** [realm/SwiftLint#5958]
### Bug Description

Sendable constraints on global-actor-qualified protocols are not redundant.

```swift
// This triggers a violation:
@MainActor protocol MyProtocol: Sendable { }
```

Minimal example showing `Sendable` is not inferred:

```swift
@MainActor protocol NotSend { }

struct S<P: NotSend>: Sendable {
    var p: P // Stored property 'p' of 'Sendable'-conforming generic struct 'S' has non-sendable type 'P'
}
```

### Environment

* SwiftLint 0.58.0
* Xcode 16.2
